[mysql] Initial pass at adding replication to the MySQL plan, plus a version bump

irvingpop opened the pull request:
this adds leader/follower setup to the MySQL plan, plus updates the version to the latest

TODO:
- [x] Add logic to support MySQL failovers to a `reconfigure` hook
- [x] squash and rebase

thesentinels commented:
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

rsertelon commented:
@irvingpop  Could you [prefix your PRs](https://github.com/habitat-sh/core-plans/blob/master/CONTRIBUTING.md#separate-changes-separate-pull-requests) with the `[packagename]` please? thanks :)

irvingpop commented:
hey @nellshamrell Yeah I will probably move that into the `README.md` file as that seems to be the convention for other plans like postgresql

nellshamrell commented:
Hi @irvingpop - have you had a moment to address the DockerFile question?

irvingpop commented:
Apologies @nellshamrell - I haven't had a chance to revisit this, it isn't functionally complete yet (I don't believe a failover would work as expected at the moment because there's no `reconfigure` hook)

Wishing for a "help needed" label to add to this PR

irvingpop commented:
hey @nellshamrell thanks for your patience.  I've added a README and moved the `docker-compose.yml` file contents there to match the postgresql plan experience.  I've also added (and lightly tested) a reconfigure hook for cluster failover.  Finally I made travis green 😄

nellshamrell commented:
Huzzah!  Thank you so much for including the docker compose file in the README - that will make testing this much easier!  I will see if I can get to reviewing/testing this today (other reviews are, of course, welcome).  If not today, then very early next week.  Ty so much for this contribution!

irvingpop commented:
Replaced by PR #1321
